### PR TITLE
[deploy] Use GH Actions to build and publish to GHCR

### PR DESCRIPTION
**Pull request by black-dragon74**
Signed-off-by: black-dragon74 <nickk.2974@gmail.com>

### CODE REVIEW

File: .github/workflows/build-and-release.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        uses: actions/checkout@v3
+
+      # Run the tests
+      - name: Run linter tests
+        run: |
+          sudo apt install shellcheck
+          sudo npm install -g dockerfile_lint
+          make
+
+      # Login to docker
```
Comment (nixpanic) [inline .github/workflows/build-and-release.yml:29]: maybe use [docker/login-action](https://github.com/docker/login-action), and [docker/build-push-action](https://github.com/docker/build-push-action) ?

**Review by nixpanic — COMMENTED** [review]
Thanks! Looks good to me. Not that this should only get merged when the `GLFS_SKEY` token is available in the repository settings.

File: .github/workflows/build-and-release.yml (hunk 1 above)
Comment (black-dragon74) [reply .github/workflows/build-and-release.yml:29]: Sure. This was my first experience with GH actions. So I went the manual route (to learn stuff). I will switch to those actions when we add multi platform builds (arm64).

Regards

**Comment by black-dragon74** [comment]
> Not that this should only get merged when the `GLFS_SKEY` token is available in the repository settings.

Tagging people who might add this to the repo.

/cc: @mscherer @deepshikhaaa @rakshithakamath94 

Regards

**Comment by black-dragon74** [comment]
> I think it would be useful to also add test on the PR, not just push to the registry.

Noted. It’ll just require a separate workflow that runs only on PRs as we’ll only need the tests and not push anything.

I have another patch in the works that’ll test the built docker images too.

Regards

**Comment by nixpanic** [comment]
On Fri, Jul 08, 2022 at 05:59:59AM -0700, Niraj Kumar Yadav wrote:
> @black-dragon74 commented on this pull request.
...
> Sure. This was my first experience with GH actions. So I went the manual route (to learn stuff). I will switch to those actions when we add multi platform builds (arm64).

Ok!

**Comment by black-dragon74** [comment]
@mscherer Can you please add the repo secret?

Regards

**Comment by nixpanic** [comment]
> I think it would be useful to also add test on the PR, not just push to the registry. Not sure what is the best way to not duplicate the code too much.

Pushing container-images was done with Travis CI earlier, so this GitHub Action looks like a good replacement. Additional workflows that test a PR would surely be nice, but that is a different task that can be done later.

**Review by nixpanic — APPROVED** [review]

**Pull request merged**